zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbAuthorizationState.java

```diff
@@ -58,26 +51,20 @@ public DbAuthorizationState(
       final ZeebeDb<ZbColumnFamilies> zeebeDb, final TransactionContext transactionContext) {
     ownerKey = new DbLong();
     resourceType = new DbString();
-    permissionType = new DbString();
-    resourceTypeAndPermissionCompositeKey = new DbCompositeKey<>(resourceType, permissionType);
-    ownerKeyAndResourceTypeAndPermissionCompositeKey =
-        new DbCompositeKey<>(ownerKey, resourceTypeAndPermissionCompositeKey);
+    ownerKeyAndResourceType = new DbCompositeKey<>(ownerKey, resourceType);
 
-    resourceIdsByOwnerKeyResourceTypeAndPermissionColumnFamily =
+    permissionsColumnFamily =
         zeebeDb.createColumnFamily(
-            ZbColumnFamilies.RESOURCE_IDS_BY_OWNER_KEY_RESOURCE_TYPE_AND_PERMISSION,
-            transactionContext,
-            ownerKeyAndResourceTypeAndPermissionCompositeKey,
-            resourceIdentifiers);
+            ZbColumnFamilies.PERMISSIONS, transactionContext, ownerKeyAndResourceType, permissions);
```

Comment:
🔧 Should we pass a separate `Permissions` instance here, to avoid the read/write buffer re-use bug?

Reply:
Why would we run into that bug here? Isn't this how always do it? E.g. in [the DbProcessState](https://github.com/camunda/camunda/blob/main/zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbProcessState.java#L125-L130)